docs/adding-new-platforms.md

```diff
@@ -0,0 +1,81 @@
+## Adding new platforms
+
+Depending on what type of platform you're adding, the instructions here will
+vary.
+
+### Adding a brand new platform
+
+- Create a new file for your mainboard in `platform-configs/`. For most
+  platforms this file will be called `[platform-vendor]-[platform-model].robot`.
+- Add the following at the top of your platform config:
+
+```robot
+*** Settings ***
+Resource    default.robot
+```
+
+- Copy the contents of `include/default.robot` to your platform config
+- Modify the file for your platform:
+    - Modify the settings appropriately for your mainboard
+    - Remove any unmodified lines - they will be sourced from `default.robot`
+- Add the platform configuration to `variables.robot:
```

Comment:
I hate the complexity and redundancy of this file, but yes, let's better document what we have and improve gradually.